al/source.cpp

```diff
@@ -3301,8 +3304,12 @@ ALsource::~ALsource()
     while(BufferList != nullptr)
     {
         std::unique_ptr<ALbufferlistitem> head{BufferList};
-        BufferList = head->mNext.load(std::memory_order_relaxed);
-        if(ALbuffer *buffer{head->mBuffer}) DecrementRef(buffer->ref);
+		/* If (BufferList->next==nullptr), the following will segfault, 
+		* thus the loading of next is moved to the next line, hopefully
+		* this won't break anything.
+		*/
+        if(ALbuffer *buffer{BufferList->mBuffer}) DecrementRef(buffer->ref);
```

Comment:
This is not needed with  the `if(ALbuffer *buffer{head->mBuffer}) DecrementRef(buffer->ref);`